src/Concerns/InteractsWithProperties.php

```diff
@@ -52,17 +52,31 @@ public function reset(...$properties)
         $freshInstance = new static;
 
         foreach ($properties as $property) {
-            $defaultValue = data_get($freshInstance, $property);
-            $unset = '__unset__';
-            $unsetByDefault = !$defaultValue && $unset === data_get($freshInstance, $property, $unset);
+            $property = str($property);
+
+            // Check if the property contains a dot which means it is actually on a nested object like a FormObject
+            if (str($property)->contains('.')) {
+                $propertyName = $property->afterLast('.');
```

Comment:
@joshhanley shouldn't it be `afterFirst` & `beforeFirst`? When I unset `a.b.c`, then it's `$component->a` that should be unset, not `$component->{'a.b'}`, no?

Reply:
@levu42 Hmm not sure I exactly follow the abc example, but if you have a form object that has a $foo property that should be unset, then first you need to get $component->form and then unset $foo on the form object.

Reply:
@joshhanley When a form has an array, let's say `$component->postForm` has a `user` property as an array with `user.name`

```
class PostForm {
    public array $user = [
        'name' => 'Foo',
        'email' => null,
    ];
}   
```

And then in the component you do `$this->reset('postForm.user.email')`, then the split should be `postForm` & `user.email`. With your code of `afterLast` & `beforeLast`, it would be split into `postForm.user` & `email`, no?

I know it's a super narrow edge case, and not sure, if it's possible at all, but I thought about it reading the code and wanted to just make sure there's not a new bug introduced quietly with this fix that then takes ages to debug in the future :)

Reply:
Yep that's how I intended it to be split, because I need to see if email is already unset or not on portForm.user. Hence I use after last as the end property I need is email. The data_get handles the postForm.user part. Does that makes sense?